enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java

```diff
@@ -314,10 +319,35 @@ public AuthenticationContext authenticate(RequestContext requestContext) throws
                     }
                     if (!"Unlimited".equals(authenticationContext.getTier())) {
                         // For subscription rate limiting, it is required to populate dynamic metadata
+                        APIConfig matchedApi = requestContext.getMatchedAPI();
+                        String apiTenantDomain = FilterUtils.getTenantDomainFromRequestURL(matchedApi.getBasePath());
+                        if (apiTenantDomain == null) {
+                            apiTenantDomain = APIConstants.SUPER_TENANT_DOMAIN_NAME;
+                        }
+                        SubscriptionDataStore datastore = SubscriptionDataHolder.getInstance()
+                                .getTenantSubscriptionStore(apiTenantDomain);
                         String subscriptionId = authenticationContext.getApiUUID() + ":" +
                                 authenticationContext.getApplicationUUID();
+                        String subPolicyName = authenticationContext.getTier();
                         requestContext.addMetadataToMap("ratelimit:subscription", subscriptionId);
-                        requestContext.addMetadataToMap("ratelimit:usage-policy", authenticationContext.getTier());
+                        requestContext.addMetadataToMap("ratelimit:usage-policy", subPolicyName);
+                        String orgList = System.getenv("CUSTOM_SUBSCRIPTION_POLICY_HANDLING_ORG");
```

Comment:
Shall we read this once, maybe as a static var? We can keep this feature flag in the top part of the file.

Reply:
We don't need to maintain this if we are using static vars.